src/renderer/dx/DxFontInfo.cpp

```diff
@@ -332,42 +335,48 @@ void DxFontInfo::SetFromEngine(const std::wstring_view familyName,
 // - dwriteFactory - The DWrite factory to use
 // Return Value:
 // - DirectWrite font collection. May be null if one cannot be created.
-[[nodiscard]] const Microsoft::WRL::ComPtr<IDWriteFontCollection1>& DxFontInfo::_NearbyCollection(gsl::not_null<IDWriteFactory1*> dwriteFactory) const
+[[nodiscard]] IDWriteFontCollection* DxFontInfo::_NearbyCollection(gsl::not_null<IDWriteFactory1*> dwriteFactory)
 {
-    // Magic static so we only attempt to grovel the hard disk once no matter how many instances
-    // of the font collection itself we require.
-    static const auto knownPaths = s_GetNearbyFonts();
+    if (_nearbyCollection)
+    {
+        return _nearbyCollection.Get();
+    }
 
     // The convenience interfaces for loading fonts from files
     // are only available on Windows 10+.
-    // Don't try to look up if below that OS version.
-    static const bool s_isWindows10OrGreater = IsWindows10OrGreater();

```

Comment:
What happened to this check? Are these available on Win7 now? (If so, we should get rid of the comment, or update to say that it doesn't matter)

Reply:
If I had to guess, it's because we switched to using `QueryInterface` to determine whether the factory supported the requisite interface.

Reply:
It is always better to do live detection of features (think backports!) instead of version number checks.

Reply:
Yeah, version checks (including the entirety of `IsWindowsXOrGreater()` functions) are basically deprecated. We're supposed to do feature checks nowadays instead.

This is especially so since the current code has a bug. Can you spot it? 😄
It's this line:
```cpp
        // Builder2 has a convenience to just feed in paths to font files.
        ::Microsoft::WRL::ComPtr<IDWriteFontSetBuilder2> fontSetBuilder2;
        THROW_IF_FAILED(fontSetBuilder.As(&fontSetBuilder2));
```

This will throw if you don't have Windows 10 Build 20348 or newer installed, which lead to #10211 and got solved "by accident" in a way, since the problem wasn't just corrupted fonts per se.

Reply:
Oh btw: Let's not store results from `VersionHelpers.h` into statics. The data it uses comes from the PEB (process environment block) and while the version check isn't trival (since you can set a ton of fields and flags when calling `VerifyVersionInfoW`), it's not a syscall and can read the version straight from process-local memory (the PEB is local to each process).